### DISCUSSION TITLE
Do nested x-transitions with :enter and :leave work?

### DISCUSSION THREAD

**Discussion by aaronstezycki** [alpinejs/alpine#4264] [1. Help] [answered]
Was creating some custom modals, and we have need for a few nested "drawers" within the modal itself.

The modal uses `x-transition:enter/enter-start/enter-end` and `x-transition:leave/leave-start/leave-end` and was looking to use the same technique within the modal itself to apply transitions to the "drawers".

e.g: 
```
<div class="modal" x-show x-transition:enter etc>
   <button @click>Drawer 1</button>
   <button @click>Drawer 2</button>
   
   <div id="drawer-1" x-show x-transition:enter etc> ... </div>
   <div id="drawer-2" x-show x-transition:enter etc> ... </div>
</div>
```

I've omitted some attribute values for brevity.

This doesn't seem to work when I first tried this out, however changing the drawers just use `x-transition` does seem to work/apply the standard alpine transitions. 

If this is current expected behaviour, a short word on the website under 'applying custom css classes' just to warn users that nested x-transition custom classes do not work would be welcome!

**Comment by SimoTod** [comment, marked as answer]
They work: https://codepen.io/SimoTod/pen/wvbPZGb?editors=1111
It's likely something in your configuration.

**Reply by aaronstezycki** [reply]
How strange! Gonna look into what seems to be me clearly getting something wrong! Thanks @SimoTod 👍🏻

**Reply by aaronstezycki** [reply]
Seems, I forgot to add transition utilities to the project! So it was applying the transforms but not using any transitions so was not animating! 

Totally my bad! Thanks again for the help :)
